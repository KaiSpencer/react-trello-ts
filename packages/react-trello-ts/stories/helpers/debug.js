export default (message) => {
  if (process.env.NODE_ENV === 'test') { return }
  if (typeof message === 'object') {
    console.dir(message)
  } else {
    console.log(message)
  }
}
