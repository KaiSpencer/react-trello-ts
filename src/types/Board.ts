import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
