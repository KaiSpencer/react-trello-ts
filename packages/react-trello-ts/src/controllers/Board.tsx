import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
