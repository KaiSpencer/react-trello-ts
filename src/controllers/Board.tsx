import classNames from "classnames";
import React, { FC, PropsWithChildren, createContext } from "react";
import { v1 } from "uuid";
import * as DefaultComponets from "../components";
import { store } from "../store/store";
import { BoardData } from "../types/Board";
import { BoardContainer } from "./BoardContainer";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			{/*
			// @ts-ignore */}
			<components.GlobalStyle />
			<BoardContainer
				id={id || v1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
