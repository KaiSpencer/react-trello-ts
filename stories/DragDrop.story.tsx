import { Meta, StoryObj } from "@storybook/react";
import React from "react";
import debug from "./helpers/debug";

import Board from "../src";

import data from "./data/base.json";

export default ({
	title: "Drag-n-Drop",
	component: Board,
	parameters: {
		info: "A demonstration of onDragStart and onDragEnd hooks for card and lanes",
	},
} satisfies Meta<typeof Board>);
type Story = StoryObj<typeof Board>;
Board.displayName = "Board";

export const DragNDrop: Story = {
	name: "Basic",
	render: (args) => {
		const handleDragStart = (cardId, laneId) => {
			debug("drag started");
			debug(`cardId: ${cardId}`);
			debug(`laneId: ${laneId}`);
		};

		const handleDragEnd = (
			cardId,
			sourceLaneId,
			targetLaneId,
			position,
			card,
		) => {
			debug("drag ended");
			debug(`cardId: ${cardId}`);
			debug(`sourceLaneId: ${sourceLaneId}`);
			debug(`targetLaneId: ${targetLaneId}`);
			debug(`newPosition: ${position}`);
			debug("cardDetails:");
			debug(card);
		};

		const handleLaneDragStart = (laneId) => {
			debug(`lane drag started for ${laneId}`);
		};

		const handleLaneDragEnd = (removedIndex, addedIndex, { id }) => {
			debug(`lane drag ended from position ${removedIndex} for laneId=${id}`);
			debug(`New lane position: ${addedIndex}`);
		};

		const shouldReceiveNewData = (nextData) => {
			debug("data has changed");
			debug(nextData);
		};

		const onCardMoveAcrossLanes = (
			fromLaneId,
			toLaneId,
			cardId,
			addedIndex,
		) => {
			debug(
				`onCardMoveAcrossLanes: ${fromLaneId}, ${toLaneId}, ${cardId}, ${addedIndex}`,
			);
		};

		return (
			<Board
				data={data}
				draggable={true}
				laneDraggable={true}
				onCardMoveAcrossLanes={onCardMoveAcrossLanes}
				onDataChange={shouldReceiveNewData}
				handleDragStart={handleDragStart}
				handleDragEnd={handleDragEnd}
				handleLaneDragStart={handleLaneDragStart}
				handleLaneDragEnd={handleLaneDragEnd}
			/>
		);
	},
};

export const DragStyling: Story = {
	render: (args) => (
		<Board
			data={data}
			cardDragClass="draggingCard"
			laneDragClass="draggingLane"
			draggable={true}
		/>
	),
};
