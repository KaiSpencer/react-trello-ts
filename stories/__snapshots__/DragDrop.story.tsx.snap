// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drag-n-Drop DragNDrop smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Planned Tasks
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                20/70
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Milk"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Buy milk
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      15 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    2 Gallons of milk at the Deli store
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Plan2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Dispose Garbage
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      10 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Sort out recyclable and waste as needed
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Plan3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Write Blog
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      30 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Can AI make memes?
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Plan4"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Pay Rent
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      5 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Transfer to bank account
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Work In Progress
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                10/20
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Wip1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Clean House
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      30 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Blocked
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                0/0
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Completed
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                2/5
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Completed1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Practice Meditation
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      15 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Use Headspace app
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Completed2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Maintain Daily Journal
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      15 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Use Spreadsheet for now
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Repeat
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Repeat1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Morning Jog
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      30 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Track using fitbit
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Archived
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Archived1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Go Trekking
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      300 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Completed 10km on cycle
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Archived2
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Archived2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Go Jogging
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      300 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Completed 10km on cycle
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Archived3
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Archived3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Go Cycling
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      300 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Completed 10km on cycle
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drag-n-Drop DragStyling smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Planned Tasks
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                20/70
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Milk"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Buy milk
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      15 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    2 Gallons of milk at the Deli store
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Plan2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Dispose Garbage
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      10 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Sort out recyclable and waste as needed
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Plan3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Write Blog
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      30 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Can AI make memes?
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Plan4"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Pay Rent
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      5 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Transfer to bank account
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Work In Progress
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                10/20
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Wip1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Clean House
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      30 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Blocked
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                0/0
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Completed
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                2/5
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Completed1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Practice Meditation
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      15 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Use Headspace app
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Completed2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Maintain Daily Journal
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      15 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Use Spreadsheet for now
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Repeat
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Repeat1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Morning Jog
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      30 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Track using fitbit
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Archived
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Archived1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Go Trekking
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      300 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Completed 10km on cycle
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Archived2
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Archived2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Go Jogging
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      300 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Completed 10km on cycle
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="width: 280px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Archived3
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span>
                1/1
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Archived3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Go Cycling
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                      300 mins
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    Completed 10km on cycle
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
